### PR TITLE
For #15937 - Remove topFrecentSite feature flag

### LINKED ISSUE #15937: Remove topFrecentSite feature flag

### PR DESCRIPTION
**Pull request by gabrielluong**
Fixes #15937. Also, removes some unused preferences and makes the customization setting more stable by not having preferences hide/show after the view is created.

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/res/values/strings.xml
Comment (gabrielluong) [reply app/src/main/res/values/strings.xml:488]: Oops, I see now that this is a placeholder and will be implemented later. Will fix this removal. Thank you for the catch!

File: app/src/main/java/org/mozilla/fenix/settings/CustomizationFragment.kt
Comment (ekager) [inline app/src/main/java/org/mozilla/fenix/settings/CustomizationFragment.kt:145]: I think we still need to keep this SharedPreferenceUpdater() bit to make sure it's updating the value in our `Settings`

**Review by ekager — CHANGES_REQUESTED** [review]
Needs one change I believe

File: app/src/main/java/org/mozilla/fenix/settings/CustomizationFragment.kt
Comment (gabrielluong) [reply app/src/main/java/org/mozilla/fenix/settings/CustomizationFragment.kt:145]: Omg thank you

**Review by ekager — APPROVED** [review]
⚡

**Comment by liuche** [comment]
but why is merging blocked???

Screen Shot 2020-10-16 at 6 31 27 PM

**Review by liuche — APPROVED** [review]

**Review by MarcLeclair — APPROVED** [review]
Just saw the change in feature flags.  I don't think this will affect performance negatively.

**Comment by MarcLeclair** [comment]
@liuche  I believe you need *all* code owners approval. I approved for the perf team. Merging should be okay now 😄

**Pull request merged**
